src/tests.c

```diff
@@ -146,22 +146,52 @@ void run_selftest_tests(void) {
     secp256k1_selftest();
 }
 
+int ecmult_gen_context_eq(const secp256k1_ecmult_gen_context *a, const secp256k1_ecmult_gen_context *b) {
+    if (a->built != b->built
+        || !secp256k1_scalar_eq(&a->blind, &b->blind)
+        || !secp256k1_gej_eq_var(&a->initial, &b->initial)) {
+        return 0;
+    }
+    return 1;
+}
+
+int context_eq(const secp256k1_context *a, const secp256k1_context *b) {
+    if (a->declassify != b->declassify
+        || !ecmult_gen_context_eq(&a->ecmult_gen_ctx, &b->ecmult_gen_ctx)
+        || a->illegal_callback.fn != b->illegal_callback.fn
+        || a->illegal_callback.data != b->illegal_callback.data
+        || a->error_callback.fn != b->error_callback.fn
+        || a->error_callback.data != b->error_callback.data) {
+        return 0;
+    }
+    return 1;
+}
+
+void test_deprecated_flags(void) {
```

Comment:
This could also test consistency of `secp256k1_context_preallocated_clone_size`

Reply:
I'm not sure how it relates to deprecated flags. If the contexts are all equal than the size of their clones is equal.